modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java

```diff
@@ -845,7 +845,7 @@ private CompletableFuture<IgniteUuid> updateAggregatedWatch() {
     private void storeEntries(Collection<IgniteBiTuple<ByteArray, byte[]>> entries, long revision) {
         appliedRevision()
                 .thenCompose(appliedRevision -> {
-                    if (revision <= appliedRevision) {
```

Comment:
Seems that `<=` is valid within given solution because all watches are registered on start only.